plugins/Plugins.cmake

```diff
@@ -73,42 +73,79 @@ MACRO(DFHACK_PLUGIN)
   CAR(PLUGIN_NAME ${PLUGIN_DEFAULT_ARGS})
   CDR(PLUGIN_SOURCES ${PLUGIN_DEFAULT_ARGS})
 
-  SET(PLUGIN_PROTOCPP)
+  SET(PLUGIN_PROTOS)
   FOREACH(pbuf ${PLUGIN_PROTOBUFS})
-    SET(PLUGIN_SOURCES ${PLUGIN_SOURCES} ${CMAKE_CURRENT_SOURCE_DIR}/proto/${pbuf}.pb.cc)
-    SET(PLUGIN_SOURCES ${PLUGIN_SOURCES} ${CMAKE_CURRENT_SOURCE_DIR}/proto/${pbuf}.proto)
-    SET(PLUGIN_PROTOCPP ${PLUGIN_PROTOCPP} ${CMAKE_CURRENT_SOURCE_DIR}/proto/${pbuf}.pb.cc)
+    LIST(APPEND PLUGIN_PROTOS ${CMAKE_CURRENT_SOURCE_DIR}/proto/${pbuf}.proto)
   ENDFOREACH()
 
-  # Tell CMake the source won't be available until build time.
-  SET_SOURCE_FILES_PROPERTIES(${PLUGIN_PROTOCPP} PROPERTIES GENERATED 1)
+  LIST(LENGTH PLUGIN_PROTOS NUM_PROTO)
```

Comment:
Is this stuff supposed to fix issues with protobuf-generated files not getting rebuilt? I'm assuming it's not warning-related. Was something wrong with what was in plugins/CMakeLists.txt? (Apparently I wrote it, although I didn't recognize it for a minute here.) I do like separating it by plugin, though.

Reply:
I rewrote it because everything that used it was using shared proto files in `/plugins/proto` which ended up meaning plugins that could actually utilize it to do something specific to the plugin were stuck making their own copy of the script.